cogito/put.go

```diff
@@ -277,14 +278,30 @@ type gitURL struct {
 	Repo  string
 }
 
+// safeUrlParse wraps [url.Parse] and returns only the error and not the URL to avoid leaking
+// passwords of the form http://user:password@example.com
+//
+// From https://github.com/golang/go/issues/53993
+func safeUrlParse(rawURL string) (*url.URL, error) {
+	parsedUrl, err := url.Parse(rawURL)
+	if err != nil {
+		if uerr := new(url.Error); errors.As(err, &uerr) {
+			// url.Parse returns a wrapped error that contains also the URL. Instead, we return only the error.
+			return nil, uerr.Err
+		}
+		return nil, errors.New("invalid URL")
+	}
+	return parsedUrl, nil
+}
+
 // parseGitPseudoURL attempts to parse rawURL as a git remote URL compatible with the
 // Github naming conventions.
 //
 // It supports the following types of git pseudo URLs:
 //   - ssh:   git@github.com:Pix4D/cogito.git; will be rewritten to the valid URL
 //     ssh://git@github.com/Pix4D/cogito.git
-//   - https: https://github.com/Pix4D/cogito.git
-//   - http:  http://github.com/Pix4D/cogito.git
+//   - https: https://github.com/Pix4D/cogito.git or https://username:password@github.com/Pix4D/cogito.git
+//   - http: http://github.com/Pix4D/cogito.git or http://username:password@github.com/Pix4D/cogito.git
```

Comment:
Could you please put each example on its own line? The versions with the u:p are different enough to be made more visible.

Reply:
Fixed in [69d2474](https://github.com/Pix4D/cogito/pull/94/commits/69d24744e01f19e6047d6393bfb78160aa816d76)